internal/controller/nnf_clientmount_controller.go

```diff
@@ -282,6 +304,50 @@ func (r *NnfClientMountReconciler) changeMount(ctx context.Context, clientMount
 	return nil
 }
 
+// Retrieve the Servers resource for the workflow and write it to a dotfile on the mount path for compute users to retrieve
+func (r *NnfClientMountReconciler) dumpServersToFile(ctx context.Context, clientMount *dwsv1alpha2.ClientMount, path string) error {
+	// Obtain the correct workflow/namespace from the Clientmount's labels
+	wf, wfFound := clientMount.Labels[dwsv1alpha2.WorkflowNameLabel]
+	ns, wfNsFound := clientMount.Labels[dwsv1alpha2.WorkflowNamespaceLabel]
+	if !wfFound || !wfNsFound {
+		return dwsv1alpha2.NewResourceError("unable to determine workflow from labels").WithMajor()
+	}
+
+	// Use the workflow/namespace labels to find the matching Server resource
+	matchLabels := dwsv1alpha2.MatchingWorkflow(&dwsv1alpha2.Workflow{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      wf,
+			Namespace: ns,
+		},
+	})
+	listOptions := []client.ListOption{
+		matchLabels,
+	}
+	serverList := &dwsv1alpha2.ServersList{}
+	if err := r.List(ctx, serverList, listOptions...); err != nil {
+		return dwsv1alpha2.NewResourceError("unable retrieve servers resource").WithError(err).WithMajor()
+	}
+
+	// Something is wrong if we found more than 1
+	if len(serverList.Items) != 1 {
+		return dwsv1alpha2.NewResourceError("wrong number of servers resources, expected 1").WithMajor()
+	}
+
+	// Dump server resource to file on mountpoint (e.g. .nnf-lustre)
+	file, err := os.Create(path)
+	if err != nil {
+		return dwsv1alpha2.NewResourceError("could not create servers file").WithError(err).WithMajor()
+	}
+	defer file.Close()
+
+	p := printers.JSONPrinter{}
+	if err := p.PrintObj(&serverList.Items[0], file); err != nil {
```

Comment:
What level of polish are we going for with this? Is this just to get something out for the tests? If we want to have this be something that lasts long term, using the servers resource as the format for the file isn't great. We wouldn't want end users to have to care about CRD api versions.

Reply:
I've flattened this out to look like the following:

```json
{
   "ost":[
      "rabbit-node-1",
      "rabbit-node-2"
   ],
   "mdt":[
      "rabbit-node-3",
      "rabbit-node-4"
   ]
}
```